extensions/s3/src/main/java/io/deephaven/extensions/s3/S3Instructions.java

```diff
@@ -110,6 +118,19 @@ public Duration readTimeout() {
      */
     @Default
     public Credentials credentials() {
+        if (profileName().isPresent() || configFilePath().isPresent() || credentialsFilePath().isPresent()) {
+            final ProfileCredentials.Builder builder = ProfileCredentials.builder();
+            profileName().ifPresent(builder::profileName);
+            if (aggregatedProfileFile != null && aggregatedProfileFile.isPresent()) {
+                builder.profileFile(aggregatedProfileFile.get());
+            }
+            if (configFilePath().isPresent() || credentialsFilePath().isPresent()) {
+                final ProfileFile result = S3Utils.aggregateProfileFile(configFilePath(), credentialsFilePath());
+                builder.profileFile(result);
+                aggregatedProfileFile = Optional.of(result);
+            }
+            return builder.build();
+        }
```

Comment:
This is an odd thing to build here; if the user builds with a specific credentials, this code is not executed. Should it be an error to explicitly set Credentials & profile / config / cred file stuff? Based on the javadoc you have for profile, it _is_ more encompassing than just credentials, right?

I guess it's possible that a user is only using the profile / config file for non-credentials based stuff.

I think a cleaner way to do this is to update our credentials building to:

```java
interface AwsSdkV2Credentials extends Credentials {

    AwsCredentialsProvider awsV2CredentialsProvider(S3Instructions instructions);
}
```

and move this profile checking into impl of `io.deephaven.extensions.s3.Credentials#defaultCredentials`. We _could_ also provide a public no-arg impl `io.deephaven.extensions.s3.Credentials#config` or something, which asserts one of these options is set on S3Instructions. I think this would also allow us to delete your version of ProfileCredentials.

Reply:
This came up slightly recently in a context where Ryan asked "now the _default_ includes _anonymous_", which makes the Deephaven default a bit different than AWS default. It _may_ make sense to have `Credentialts.defaultCredentials()` be the _Deephaven_ default (which checks profile stuff, does the default chain, and falls back to anonymous), and introduce an _AWS_ default.